server/src/main/scala/de/frosner/broccoli/instances/InstanceModule.scala

```
@@ -0,0 +1,70 @@
+
+import java.nio.file.FileSystems
+import javax.inject.Singleton
+
+import com.google.inject.{AbstractModule, Provides}
+import de.frosner.broccoli.instances.conf.InstanceConfiguration
+import de.frosner.broccoli.util.Logging
+import net.codingwell.scalaguice.ScalaModule
+import play.api.Configuration
+import play.api.inject.ApplicationLifecycle
+import play.api.libs.ws.WSClient
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Provide Instance and InstanceStorage configuration from the Play configuration
+  */
+class InstanceModule extends AbstractModule with ScalaModule with Logging {
+  override def configure(): Unit = {}
+
+  @Provides
+  @Singleton
+  def provideInstanceConfiguration(config: Configuration): InstanceConfiguration =
+    InstanceConfiguration.fromConfig(config.underlying.getConfig("broccoli.instances"))
+
+  @Provides
+  @Singleton
+  def provideInstanceStorage(instanceConfig: InstanceConfiguration,
+                             ws: WSClient,
+                             applicationLifecycle: ApplicationLifecycle): InstanceStorage = {
+    val maybeInstanceStorage = instanceConfig.storageConfiguration.storageType match {
+      case StorageType.FileSystem => {
+        val config = instanceConfig.storageConfiguration.fsConfig
+        val path = FileSystems.getDefault.getPath(config.url).toAbsolutePath
+        Try(FileSystemInstanceStorage(path.toFile))
+      }
+      case StorageType.CouchDB => {
+        val config = instanceConfig.storageConfiguration.couchDBConfig
+        Try(CouchDBInstanceStorage(config.url, config.dbName, ws))
+      }
+      case invalidStorageType => throw new IllegalArgumentException("Wrong instance storage type")
+    }
+
+    val instanceStorage = maybeInstanceStorage match {
+      case Success(storage) => storage
+      case Failure(throwable) =>
+        Logger.error(s"Cannot start instance storage: $throwable")
+        throw throwable
+    }
```

I'm not sure that we really need to use `Try` here given that we just re-throw the exception.

I'd just remove `Try`, create the instance storages directly and let exceptions propagate.  They'll end up in the logger anyway.